src/global_context.hpp

```diff
@@ -92,25 +90,27 @@ class GlobalContext {
     /**
      * @brief submit task to thread pool
      */
-    template <class... Args>
+    template <typename F, typename R = std::invoke_result_t<std::decay_t<F>>>
     auto
-    submit(Args&&... args) {
-        return thread_pool_.submit(std::forward<Args>(args)...);
+    submit_task(F&& function) {
+        return thread_pool_.submit_task(function);
     }
 
     /**
      * @brief push task to thread pool
      */
-    template <class... Args>
+    template <class F>
     void
-    push_task(Args&&... args) {
-        return thread_pool_.push_task(std::forward<Args>(args)...);
+    push_task(F&& function) {
+        thread_pool_.detach_task(function);
     }
 
+    /*
```

Comment:
Why is this commented out?

Reply:
I was considering removing it. I'm not sure if I would any caller to be able to block for all queued tasks. I might want someone to be able to call it so I might leave it commented out.